packages/server/src/api/controllers/query/index.ts

```diff
@@ -142,6 +142,78 @@ export async function preview(ctx: UserCtx) {
 
   const authConfigCtx: any = getAuthConfig(ctx)
 
+  function getFieldMetadata(
+    field: any,
+    key: string,
+    nestedSchemaFields: {
+      [key: string]: Record<string, string | QuerySchema>
+    }
+  ): QuerySchema {
+    const makeQuerySchema = (
+      type: FieldType,
+      name: string,
+      subtype?: string
+    ): QuerySchema => ({
+      type,
+      name,
+      subtype,
+    })
+    let type = typeof field,
+      fieldMetadata = makeQuerySchema(FieldType.STRING, key)
+    if (field)
+      switch (type) {
+        case "boolean":
+          fieldMetadata = makeQuerySchema(FieldType.BOOLEAN, key)
+          break
+        case "object":
+          if (field instanceof Date) {
+            fieldMetadata = makeQuerySchema(FieldType.DATETIME, key)
+          } else if (Array.isArray(field)) {
+            const firstFiveItems = field.slice(0, 5)
```

Comment:
wouldn't use just the first five - would bump this to full data set as its easier to understand what is happening rather than explaining its just the first five.

Reply:
As discussed, removed the limit and performing the schema build across all rows. Didn't appear to have any noticeable performance impact.